### ISSUE TITLE
Phone volume buttons can't control the bluetooth with VoiceComm

### ISSUE THREAD

**Issue by hpsaturn** [google/oboe#1997]
Android version(s): oriole:14/AP1A.240305.019.A1/11445699:user
Android device(s): Pixel6 stock firmware
Oboe version: 1.8.1 
App name used for testing: OboeTester(81) v2.5.10 (debug version from git tag 1.8-stable)
 
**Description**

The phone device when is connected to the Bluetooth headset, can't control the volume from its volume buttons.

**Steps to reproduce**

1. Connect to the Bluetooth headset
2. Open OboeTester in TestOutput
3. Choose: Usage->VoiceComm, ContentType->Speech
4. Open, Start (confirm the audio output in the headset)
5. Try to change the volume from the phone buttons

![screenshot20240409_115600](https://github.com/google/oboe/assets/423856/2a931e69-59ef-49d4-8554-2e0257d2fb35)

**Expected behavior**

Changing the headset volume from the phone volume buttons.

**Actual behavior**

The volume bars are showed but you are not able to change the volume. If you expand the volume settings UI is possible, but the phone choose the wrong volume bar (media volume instead the call volume bar).  From the headset buttons, you are able to controlling it, but the device choose the wrong volume bar (media instead the call bar).

![screenshot20240409_115507](https://github.com/google/oboe/assets/423856/b28d3462-dbbd-42a0-a020-68f05e7c9f77)  
(Headset volume buttons control)

![screenshot20240409_122316](https://github.com/google/oboe/assets/423856/96af86bb-1c84-46bd-80b2-e20e4d43569a)  
(Device volume buttons control with audio settings expanded)

**Device**

```bash
ro.product.brand = google
ro.product.manufacturer = Google
ro.product.model = Pixel 6
ro.product.device = oriole
ro.product.cpu.abi = arm64-v8a
ro.build.description = oriole-user 14 AP1A.240305.019.A1 11445699 release-keys
ro.hardware = oriole
ro.hardware.chipname = 
ro.arch = 
| grep aaudio = [aaudio.hw_burst_min_usec]: [2000]
[aaudio.mmap_exclusive_policy]: [2]
[aaudio.mmap_policy]: [2]
```

**Any additional context**

I believe that this issue has relation with the closed issue #1814. 
Also when you do that with the release verion of OboeTester from GooglePlay, (Oboe 1.7.1), we can controlling the headset volume from the device, but it choose the wrong bars, but is acceptable. I guess, that the Oboe 1.8.1 introduces a problem, or the current stock firmware of Android. Thanks in advance.

**Comment by philburk** [comment]
Thanks for reporting this.

> I guess, that the Oboe 1.8.1 introduces a problem,

We added code that calls setVolumeControlStream() based on the stream attributes.
You can disable that code by unchecking a checkbox near the bottom of the TEST OUTPUT screen.
The checkbox may be called "VolumeByAttr" or "SetStreamByAttr".

That may switch between the two behaviors you are seeing.

I believe this matches an internal bug:
[b/285185485](http://b/285185485) | P2 | Volume keys do not work when Usage:VoiceCommunication and ContentType:Speech

There was a regression from Android 11.

That has been fixed and the fix will be available in a future release.

**Comment by philburk** [comment]
@robertwu1 - can you please verify the fix in ag/26571511
and close this Issue if there are no loose ends.

**Comment by robertwu1** [comment]
With ag/26571511, this works if Bluetooth SCO is connected via the BT_SCO checkbox at the bottom of the Test Output screen. Thanks for reporting this! I am closing this PR as the fix seems to work.
